src/lib/svg2component.js

```diff
@@ -11,23 +11,56 @@ module.exports = (name, svg, isTypeScriptOutput) => {
   const $svg = $('svg');
   toReactAttributes($svg, $);
   const children = $svg.html();
-  const viewBox = $svg.attr('viewBox');
-  const iconJsx = `<Icon viewBox="${viewBox}" {...props}>${children}</Icon>`;
+  const viewBox = $svg.attr('viewBox') ? `viewBox="${$svg.attr('viewBox')}"` : '';
   const code = isTypeScriptOutput ?
     `
       import * as React from 'react';
-      import Icon, {IconProps} from '../Icon';
-      const ${name}: React.SFC<IconProps> = props => (
-        ${iconJsx}
+      export interface ${name}Props extends React.SVGAttributes<SVGElement> {
+        size?: string;
+        width: string;
+        height: string;
+      }
+      const widthFromSvg = ${$svg.attr('width')} || '1em';
+      const heightFromSvg = ${$svg.attr('width')} || '1em';
+      const ${name}: React.SFC<${name}Props> = props => (
+        <svg
+          ${viewBox}
+          width={props.size || widthFromSvg}
+          height={props.size || heightFromSvg}
+          {...props}
+        >
+          ${children}
+        </svg>
       );
+      ${name}.defaultProps = {
+        fill: 'currentColor'
```

Comment:
Just a suggestion: I think it's a bit cleaner to simply add `fill="currentColor"` above as an svg attribute and remove defaultProps.